src/software/ai/hl/stp/tactic/goalie/goalie_fsm.h

```diff
@@ -0,0 +1,429 @@
+         * @param processEvent processes the DribbleFSM::Update
+         */
+        const auto update_dribble = [](auto event,
+                                       back::process<DribbleFSM::Update> processEvent) {
+            double clear_origin_x =
+                getNoChipRectangle(event.common.world.field()).xMax() +
+                ROBOT_MAX_RADIUS_METERS;
+            Point clear_origin =
+                Point(clear_origin_x, event.common.world.ball().position().y());
+
+            Angle clear_direction = (event.common.world.ball().position() -
+                                     event.common.world.field().friendlyGoalCenter())
+                                        .orientation();
+
+            DribbleFSM::ControlParams control_params{
+                .dribble_destination       = clear_origin,
+                .final_dribble_orientation = clear_direction,
+                .allow_excessive_dribbling = false,
+            };
+
+            // update the dribble fsm
+            processEvent(DribbleFSM::Update(control_params, event.common));
+        };
+
+        /**
+         * Action that updates the MoveIntent to position the goalie in the best spot to
+         * block shots.
+         *
+         * @param event GoalieFSM::Update event
+         */
+        const auto update_position_to_block = [](auto event) {
+            Point goalie_pos = getGoaliePositionToBlock(
+                event.common.world.ball(), event.common.world.field(),
+                event.control_params.goalie_tactic_config);
+            Angle goalie_orientation =
+                (event.common.world.ball().position() - goalie_pos).orientation();
+
+            // what should the final goalie speed be, so that the goalie accelerates
+            // faster
+            auto goalie_final_speed =
+                event.control_params.goalie_tactic_config->getGoalieFinalSpeed()->value();
+
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), goalie_pos, goalie_orientation,
+                goalie_final_speed, DribblerMode::OFF, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::AUTOCHIP, YEET_CHIP_DISTANCE_METERS},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *position_to_block_s + update_e[should_panic] / update_panic    = panic_s,
+            position_to_block_s + update_e[should_dribble] / update_dribble = dribble_s,
+            position_to_block_s + update_e[should_chip] / update_chip       = chip_s,
+            position_to_block_s + update_e / update_position_to_block,
+            panic_s + update_e[should_chip] / update_chip = chip_s,
+            panic_s + update_e[panic_done] = X, panic_s + update_e / update_panic,
+            dribble_s + update_e / update_dribble, dribble_s = chip_s,
```

Comment:
Can you explain line 333 and 334?

Reply:
Line 333 and 334 each have two transitions on the same line: 
Line 333: the first transition is from panic_s to X, if on update_e the ball is no longer an immediate threat (panic_done); the second transition is the internal transition to keep updating the panic_s
Line 334: the first transition is the internal transition to keep updating dribble_s; the second is to transition to chip_s once dribble_s has completed - we don't need a separate guard here to determine this bc it will naturally transition once the dribble is complete (as determined by the dribble fsm)